apple/Sources/FerrostarMapLibreUI/Extensions/View.swift

```diff
@@ -0,0 +1,13 @@
+import SwiftUICore
+
+extension View {
+    /// Do something reasonable-ish for clients that don't yet support
+    /// safeAreaPadding - in this case, fall back to regular padding.
+    func safeAreaPaddingPolyfill(_ insets: EdgeInsets) -> AnyView {
```

Comment:
I'm not sure if there's existing patterns for polyfill type methods, but since it gets used in a few places, it seemed worth extracting. Otherwise, doing an `#availability` check inline in a view body is annoying because you have to do an assignment.

like:

```swift
var body: some View {
  HStack {
    VStack {
      Text("Foo")
    }.safeAreaPaddingPolyfill(16)
  }
}
```

Has to become something like:
```swift
var body: some View {
  HStack {
    let foo = VStack {
      Text("Foo")
    }
    if #available(iOS 17.0, *) {
      foo.safeAreaPadding(insets)
    } else {
      foo.padding(insets)
    }
  }
}
```

Reply:
Looks good to me :) There isn't an existing pattern, partly because we explicitly started with a fairly recent support target (I actually wanted to start with 17 so we could get the `@Observable` macro, but @Archdoog correctly pointed out that this would unnecessarily limit adoption haha).